src/arch_ppc.py

```diff
@@ -661,17 +764,31 @@ def make_memory_access(arg: Argument) -> Access:
             inputs=inputs,
             clobbers=clobbers,
             outputs=outputs,
+            reads_memory=reads_memory,
+            writes_memory=writes_memory,
             jump_target=jump_target,
             function_target=function_target,
             is_conditional=is_conditional,
             is_return=is_return,
         )
 
+    def simplify_ir(self, flow_graph: FlowGraph) -> None:
+        simplify_ir_patterns(self, flow_graph, self.ir_patterns)
```

Comment:
I can't move this to translate.py, if I did then it couldn't be called from flow_graph.py in `build_flowgraph`.
I don't want to call it from `translate_to_ast`, because that function can be called multiple times for the same flow graph.

I could leave it in ir_patterns.py, and call `simplify_ir_patterns` from main.py?

Reply:
I was thinking keeping the declaration in flow_graph.py, but defining it in Arch in translate.py. Moving to main is also an option